common/prometheus-server/templates/alerts/_prometheus.alerts.tpl

```diff
@@ -21,145 +21,284 @@ groups:
+  - alert: PrometheusTargetSyncFailure
+    expr: |
+      increase(prometheus_target_sync_failed_total{prometheus="{{ include "prometheus.name" . }}"}[30m]) > 0
+    for: 5m
     labels:
-      context: availability
       service: {{ default "metrics" $root.Values.alerts.service }}
       support_group: {{ default "observability" $root.Values.alerts.support_group }}
-      severity: info
-      playbook: 'docs/support/playbook/prometheus/failed_scrapes.html'
-      meta: 'Prometheus `{{`{{ $labels.prometheus }}`}}` fails to scrape targets'
+      severity: critical
+      playbook: docs/support/playbook/prometheus/failed_scrapes.html
+      meta: Prometheus `{{`{{ $labels.prometheus }}`}}` has failed to sync targets.
     annotations:
-      description: 'Prometheus `{{`{{ $labels.prometheus }}`}}` has many scrapes that exceed the sample limit'
-      summary: Prometheus fails to scrape targets.
+      description: |
+        `{{`{{ printf "%.0f" $value }}`}} targets in Prometheus `{{`{{ $labels.prometheus }}`}}`
+        have failed to sync because invalid configuration was supplied.
+      summary: Prometheus has failed to sync targets.
+
+  - alert: PrometheusHighQueryLoad
+    expr: |
+      avg_over_time(prometheus_engine_queries{prometheus="{{ include "prometheus.name" . }}"}[5m])
+      /
+      max_over_time(prometheus_engine_queries_concurrent_max{prometheus="{{ include "prometheus.name" . }}"}[5m])
+      >
+      0.8
+    for: 15m
+    labels:
+      service: {{ default "metrics" $root.Values.alerts.service }}
+      support_group: {{ default "observability" $root.Values.alerts.support_group }}
+      severity: critical
+      playbook: docs/support/playbook/prometheus/failed_scrapes.html
+      meta: Prometheus `{{`{{ $labels.prometheus }}`}}` is reaching its maximum capacity serving concurrent requests.
+    annotations:
+      description: |
+        Prometheus `{{`{{ $labels.prometheus }}`}}` query API
+        has less than 20% available capacity in its query
+        engine for the last 15 minutes.
+      summary: Prometheus is reaching its maximum capacity serving concurrent requests.
 
   {{- if $root.Values.alerts.multipleTargetScrapes.enabled }}
   - alert: PrometheusMultipleTargetScrapes
-    # we exclude the following:
-    # * cadvisor metrics because it has the same instance as the kubelet but a different path
-    # e.g. 10.246.204.80:10250/metrics vs. 10.246.204.80:10250/metrics/cadvisor
-    # * pod service discovery job, we have a dedicated alert for that
-    expr: sum by (job, ccloud_support_group) (up * on(instance, cluster) group_left() (sum by(instance, cluster) (up{job !~ "kubernetes-cadvisors|kubernetes-kubelet|.*-pod-sd|{{ $root.Values.alerts.multipleTargetScrapes.exceptions | join "|" }}"}) > 1))
+    expr: sum by (job, ccloud_support_group) (up * on(instance, cluster) group_left() (sum by(instance, cluster) (up{job!~"{{ $root.Values.alerts.multipleTargetScrapes.exceptions | join "|" }}"}) > 1))
```

Comment:
What about the previously excluded jobs? Aren't they around anymore?

Reply:
This is only needed for kube-state-metrics. Therefore, this must be excluded in the specific Prometheus, not in the common chart. https://github.com/sapcc/helm-charts/blob/master/common/prometheus-server/values.yaml#L395-L398

Reply:
I will create a pull request on the affected Prometheus charts to regain the exceptions.